transport/tlstcp/tlstcp.go

```diff
@@ -193,10 +200,8 @@ func (t *tlsTran) NewDialer(addr string, sock mangos.Socket) (mangos.PipeDialer,
 		return nil, err
 	}
 
-	d := &dialer{sock: sock, opts: newOptions(t)}
-	if d.addr, err = mangos.ResolveTCPAddr(addr); err != nil {
```

Comment:
Same comment as above -- I think we want at least one of these resolutions to be done synchronously for the caller.